pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -359,8 +360,19 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 					return err
 				}
 				return nil
+			case storagehelpers.GetPersistentVolumeClaimClass(claim) == "":
+				if storageClasses, err := ctrl.kubeClient.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{}); err != nil {
+					for _, sc := range storageClasses.Items {
+						if sutil.IsDefaultAnnotation(sc.ObjectMeta) {
+							claim.Spec.StorageClassName = &sc.Name
+							ctrl.updateClaim(claim)
+							return nil
+						}
+					}
+				}
```

Comment:
Please extract all this getting default SC and modifying PV into a separate function and just call it. I like the giant `if-then-else` in `syncUnboundClaim` as clean as possible.